base/Dockerfile

```diff
@@ -37,15 +37,28 @@ ENV LANG="C.UTF-8"
 # Set the classpath for JARs required by `cub`
 ENV CUB_CLASSPATH='"/usr/share/java/cp-base-new/*"'
 
+# Zulu openJDK
+ENV ZULU_OPENJDK="zulu8.40.0.25-ca-jdk8.0.222-linux_x64"
+
 COPY requirements.txt .
 
-RUN apt update \
-    && apt install -y bash curl git wget netcat-openbsd python python-pip python-setuptools python-enum34 build-essential libssl-dev libffi-dev python-dev\
-    && pip install --install-option="--prefix=/usr/local" --upgrade -rrequirements.txt \
-    && apt remove --purge -y git python-pip \
-    && apt autoremove -y \
-    && rm -rf /var/lib/apt/lists/*     \
-    && rm requirements.txt \
+RUN microdnf install yum \
+    && yum update -q -y \
+    && yum install -y git wget nc python3 gcc libffi-devel redhat-rpm-config python3-devel openssl-devel make tar \
+    && pip3 install --install-option="--prefix=/usr/local" --upgrade -rrequirements.txt \
+    && mkdir -p /usr/lib/jvm \
+    && cd /usr/lib/jvm \
+    && curl -O https://cdn.azul.com/zulu/bin/${ZULU_OPENJDK}.tar.gz \
+    && tar -xzvf ${ZULU_OPENJDK}.tar.gz \
+    && rm -rf ${ZULU_OPENJDK}.tar.gz \
+    && cd / \
+    && ln -s /usr/lib/jvm/${ZULU_OPENJDK}/bin/java /usr/bin/java \
+    && export JAVA_HOME=/usr/lib/jvm/${ZULU_OPENJDK} \
+    && export PATH=$JAVA_HOME/bin:$PATH \
+    && yum remove -y git \
```

Comment:
do you know why we remove pip on the current base image? not saying we should be doing on that on this new image, just kind of wondering why.

Reply:
since pip is part of python, I do not think we supposed to remove python.